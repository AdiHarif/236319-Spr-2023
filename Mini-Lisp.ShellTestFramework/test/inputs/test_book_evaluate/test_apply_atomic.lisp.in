; =================================================
; apply-atomic - 1 argument functions:
; =================================================
(apply-atomic 'atom '(a NIL) '(a.x t.t nil.nil))      ; T
(apply-atomic 'atom '(a NIL) '(a.(x) t.t nil.nil))    ; NIL
(apply-atomic 'car '(a NIL) '(a.(x) t.t nil.nil))     ; X
(apply-atomic 'car '(a NIL) '(a.(x y z) t.t nil.nil)) ; X
(apply-atomic 'car '(a NIL) '(a.(x.y) t.t nil.nil))   ; X.Y
(apply-atomic 'cdr '(a NIL) '(a.(x) t.t nil.nil))     ; NIL
(apply-atomic 'cdr '(a NIL) '(a.(x y z) t.t nil.nil)) ; (Y Z)
(apply-atomic 'cdr '(a NIL) '(a.(x.y) t.t nil.nil))   ; NIL

; =================================================
; apply-atomic - 2 argument functions:
; =================================================
(apply-atomic 'cons '(a b) '(a.x b.y t.t nil.nil))  ; (X.Y)
(apply-atomic 'cons '(a b) '(a.x b.x t.t nil.nil))  ; (X.X)
(apply-atomic 'eq '(a a) '(a.x t.t nil.nil))        ; T
(apply-atomic 'eq '(a b) '(a.x b.y t.t nil.nil))    ; NIL
(apply-atomic 'eq '(a b) '(a.x b.x t.t nil.nil))    ; T
(apply-atomic 'set '(a b) '(a.x b.y t.t nil.nil))   ; Y

; =================================================
; apply-atomic - cond:
; =================================================
(apply-atomic 'cond '(()) '(t.t nil.nil))                      ; NIL
(apply-atomic 'cond '((t nil)) '(t.t nil.nil))                 ; NIL
(apply-atomic 'cond '((a t)) '(a.t t.t nil.nil))               ; T
(apply-atomic 'cond '((a b)) '(a.t b.t t.t nil.nil))           ; T
(apply-atomic 'cond '((t a) (nil b)) '(a.a b.b t.t nil.nil))   ; A
(apply-atomic 'cond '((nil a) (t b)) '(a.a b.b t.t nil.nil))   ; B
(apply-atomic 'cond '((a b) (b a)) '(a.t b.t t.t nil.nil))     ; T
(apply-atomic 'cond '((a b) (b a)) '(a.nil b.t t.t nil.nil))   ; NIL
(apply-atomic 'cond '((a b) (a b) (b a)) '(a.nil b.t t.t nil.nil)) ; NIL

; =================================================
; apply-atomic - the error function:
; =================================================
(apply-atomic 'error '(NIL) '(t.t nil.nil)) ; NIL
(apply-atomic 'error '(a) '(a.my_err t.t nil.nil)) ; MY_ERR

; =================================================
; apply-atomic - Errors:
; =================================================
(apply-atomic 'bla '(a b) '(a.x b.y t.t nil.nil))   ; error
