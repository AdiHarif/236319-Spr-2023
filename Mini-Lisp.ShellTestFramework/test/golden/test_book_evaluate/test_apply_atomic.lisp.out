T
NIL
X
X
(X.Y)
NIL
(Y Z)
NIL
(X.Y)
(X.X)
T
NIL
T
Y
NIL
NIL
T
T
A
B
T
NIL
NIL
Traceback (most recent call last):
	From function APPLY-ATOMIC:
		Args: (BLA (A B) ((A.X) (B.Y) (T.T) (NIL)))
	From function COND:
		Args: (((EQ (QUOTE COND) ATOMIC) (EVALUATE-COND ACTUALS A-LIST)) (T (APPLY-EAGER-ATOMIC ATOMIC (EVALUATE-LIST ACTUALS A-LIST) A-LIST)))
	From function APPLY-EAGER-ATOMIC:
		Args: (BLA (X Y) ((A.X) (B.Y) (T.T) (NIL)))
	From function COND:
		Args: (((EQ ATOMIC (QUOTE ERROR)) (ERROR ACTUALS)) ((EQ ATOMIC (QUOTE EVAL)) (EVALUATE (CAR ACTUALS) A-LIST)) (T (APPLY-TRIVIAL-ATOMIC ATOMIC (CAR ACTUALS) (CAR (CDR ACTUALS)))))
	From function APPLY-TRIVIAL-ATOMIC:
		Args: (BLA X Y)
	From function COND:
		Args: (((EQ ATOMIC (QUOTE ATOM)) (ATOM FIRST)) ((EQ ATOMIC (QUOTE CAR)) (CAR FIRST)) ((EQ ATOMIC (QUOTE CDR)) (CDR FIRST)) ((EQ ATOMIC (QUOTE CONS)) (CONS FIRST SECOND)) ((EQ ATOMIC (QUOTE EQ)) (EQ FIRST SECOND)) ((EQ ATOMIC (QUOTE SET)) (SET FIRST SECOND)) (T (ERROR (QUOTE SOMETHING-WENT-WRONG) ATOMIC)))
	From function ERROR:
		Args: SOMETHING-WENT-WRONG
Error [SOMETHING-WENT-WRONG] in (ERROR (QUOTE SOMETHING-WENT-WRONG) ATOMIC)
